pkg/controller/periodic/periodic.go

```diff
@@ -115,32 +115,45 @@ func (c *Controller) Gather() {
 
 	allFunctionReports := make(map[string]gather.GathererFunctionReport)
 
-	for _, gatherer := range gatherersToProcess {
-		func() {
-			name := gatherer.GetName()
-			start := time.Now()
-
-			ctx, cancel := context.WithTimeout(context.Background(), c.configurator.Config().Interval/2)
-			defer cancel()
+	configChangedChannel, configChangedCloser := c.configurator.ConfigChanged()
+	defer configChangedCloser()
+
+	// TODO: use a separate field in the config for timeout
+	ctx, cancel := context.WithTimeout(context.Background(), c.configurator.Config().Interval/2)
+	defer cancel()
+	go func() {
+		// once the configChangedChannel is closed by configChangedCloser
+		// (which happens when the Gather function finishes) this goroutine will stop
+		for range configChangedChannel {
+			// TODO: check that our field has changed
+			fmt.Println()
+			fmt.Println("!!! CONFIG WAS CHANGED DURING GATHERING !!!")
+			fmt.Println("cancelling the gathering...")
+			fmt.Println()
+			cancel()
```

Comment:
So this is sufficient to stop the gathering in the progress right? :)

Reply:
We also need to stop the workers and so on to stop it even faster, but essentially, yes